### ISSUE TITLE
Audit Premium Events SharePoint & Exchange

### ISSUE THREAD

**Issue by Sloane4** [cisagov/ScubaGear#308]
# 💡 Summary #

Direct the user to enable 2 audit events through admin power shell to enable specific ability for Incident Response.
https://learn.microsoft.com/en-us/microsoft-365/compliance/audit-premium-setup?view=o365-worldwide#step-2-enable-audit-premium-events

## Motivation and context ##

Audit (Premium) features such as the ability to log crucial events such as MailItemsAccessed and Send require an appropriate E5 license assigned to users. Additionally, the Advanced Auditing app/service plan must be enabled for those users. You have to enable two Audit (Premium) events (SearchQueryInitiatedExchange and SearchQueryInitiatedSharePoint) to be logged when users perform searches in Exchange Online and SharePoint Online. In additional to the default policy that retains Azure AD, Exchange, OneDrive, and SharePoint audit records for one year, you can create additional audit log retention policies to meet the requirements of your organization's security operations, IT, and compliance teams. Once enabled you can search the audit log for Audit (Premium) events and other activities during forensic investigations of compromised accounts and other types of security or compliance investigations.

## Implementation notes ##

- Add new policy to Exchange baseline
- Add new policy to SharePoint baseline
- Create non implemented check in rego
- Update unit test cases for new policy
- Research if there is a way to check if audit events can be checked through SCuBA
- IF audit events can be checked open a new issue to enhance the code

## Acceptance criteria ##

- [ ] Updated policy has the change noted in implementation notes
- [ ] Non-Implementation check in Exchange & SharePoint
- [ ] All unit test cases pass

**Comment by Sloane4** [comment]
I argue that this issue is dependent on the decision of #88. The previous issue asks if we should & the creation of an appropriate policy (should vs shall). While this issue focuses more on what to do if we add advanced audit logs (i.e. updating the baseline, code, testing). I would agree that this issue can be updated to better reflect that, but I see them as distinct issues.

**Comment by tkol2022** [comment]
@Sloane4 @buidav  Can you please test the instructions you hyperlinked above to add the SearchQueryInitiatedSharePoint and SearchQueryInitiatedExchange audit event against a couple of our tenants that have the required license and then share your results here?

I'd like to make sure:

1. We flush out any nuances to the policy language.
2. We record the dependencies for these events to work. For example, it seems like a tenant user must have the MS 365 Advanced Auditing license turned on before an Administrator can enable the SearchQueryInitiated events noted above (see screenshot below)
3. We record the steps needed for an Administrator to enable these events.

Also, I have a question for you: I think it would make sense to write a single version of this policy and then use the same language across the EXO and Sharepoint/OneDrive baselines since the only difference between the two is the audit event being configured. In other words we want consistency in the policy language. Do you agree? 

![image](https://github.com/cisagov/ScubaGear/assets/107076927/f130c1d7-d270-4228-abae-90e546eef275)

**Comment by tkol2022** [comment]
@schrolla @ahuynhMITRE Noting this issue as a potential candidate for deferral to Flipper if necessary. This is given that the result of this issue may spark a baseline policy addition to a couple of baselines and where we are right now with the status of the baseline reviews.

**Comment by Sloane4** [comment]
Moved to blocked as we are taking a wait & see approach for Microsoft's changes to the advanced auditing licensing

**Comment by tkol2022** [comment]
Kindly update this issue to state specifically what is blocking it and an estimated unblock date. If we are unsure when it will become unblocked, I would consider removing it from Glacier or assigning a cutoff date after which it will be pushed to the next release.

**Comment by ahuynhMITRE** [comment]
@twneale is this issue still expect to have baseline changes and if yes what is the expected date to have these changes into a PR and reviewed?
